### PR TITLE
Avoid implicit conversions in Ruby extension code

### PR DESCRIPTION
**Pull request by marcinwyszynski**
This fixes outstanding issues with https://github.com/grpc/grpc/pull/2476

### CODE REVIEW

**Comment by marcinwyszynski** [comment]
I signed it!

File: src/ruby/ext/grpc/rb_grpc.c [hunk 1]
```diff
@@ -139,7 +139,7 @@ gpr_timespec grpc_rb_time_timeval(VALUE time, int interval) {
           rb_raise(rb_eRangeError, "%f out of Time range",
                    RFLOAT_VALUE(time));
         }
-        t.tv_nsec = (time_t)(d * 1e9 + 0.5);
+        t.tv_nsec = (int)(time_t)(d * 1e9 + 0.5);
```
Comment (murgatroid99) [inline src/ruby/ext/grpc/rb_grpc.c]: I don't think this needs to be a double conversion.

Comment (marcinwyszynski) [inline src/ruby/ext/grpc/rb_grpc.c]: Fair enough, me dumb, fixed :)

**Comment by nicolasnoble** [comment]
This is ok to test.

File: src/ruby/ext/grpc/rb_call.c [hunk 2]
```diff
@@ -235,7 +235,7 @@ static VALUE grpc_rb_call_set_metadata(VALUE self, VALUE metadata) {
 */
 static int grpc_rb_md_ary_fill_hash_cb(VALUE key, VALUE val, VALUE md_ary_obj) {
   grpc_metadata_array *md_ary = NULL;
-  int array_length;
+  long array_length;
   int i;
```
Comment (soltanmm) [inline src/ruby/ext/grpc/rb_call.c]: This counter is incremented up to `array_length` and used as the index argument to `rb_ary_entry`. Would you mind changing this one as well?

Comment (marcinwyszynski) [inline src/ruby/ext/grpc/rb_call.c]: No bother, done.

**Comment by marcinwyszynski** [comment]
PTAL

**Comment by marcinwyszynski** [comment]
Ping?

**Comment by murgatroid99** [comment]
OK, the Jenkins tests passed.

**Comment by marcinwyszynski** [comment]
Thanks folks.

**Pull request merged**